compiler/rustc_hir/src/lang_items.rs

```diff
@@ -103,13 +109,13 @@ macro_rules! language_item_table {
                 self.items[it as usize].ok_or_else(|| format!("requires `{}` lang_item", it.name()))
             }
 
+            /// Returns the [`DefId`]s of all lang items in a group.
             pub fn group(&self, group: LangItemGroup) -> &[DefId] {
                 self.groups[group as usize].as_ref()
             }
 
             $(
-                /// Returns the corresponding `DefId` for the lang item if it
-                /// exists.
+                #[doc = concat!("Returns the [`DefId`] of the `", stringify!($name), "` lang item if it is defined.")]
                 #[allow(dead_code)]
```

Comment:
What happens if you remove this `allow`?

Reply:
Did you ever find out what happens?

Reply:
It wasn't necessary, I just removed it. Good catch!